controller/controller.go

```diff
@@ -0,0 +1,191 @@
+// Package controller implements a controller client providing basic operations against a
+// controller endpoint through controller's HTTP API. Higher level scheduling and
+// management should be built on top of that.
+package controller
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/giantswarm/formica/fleet"
+)
+
+type Config struct {
+	Fleet fleet.Fleet
+}
+
+func DefaultConfig() Config {
+	newFleetConfig := fleet.DefaultConfig()
+	newFleet, err := fleet.NewFleet(newFleetConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	newConfig := Config{
+		Fleet: newFleet,
+	}
+
+	return newConfig
+}
+
+// Controller defines the interface a controller needs to implement to provide
+// operations for groups of unit files against a fleet cluster.
+type Controller interface {
+	// Submit schedules a group on the configured fleet cluster. This is done by
+	// setting the state of the units in the group to loaded.
+	Submit(req Request) error
+
+	// Start starts a group on the configured fleet cluster. This is done by
+	// setting the state of the units in the group to launched.
+	Start(req Request) error
```

Comment:
No we don't. I just thought that it is nearly the same interface. I can create a new request type without the content. No strong feelings.